packages/core/src/modules/connections/handlers/ConnectionRequestHandler.ts

```diff
@@ -9,23 +10,38 @@ import { ConnectionRequestMessage } from '../messages'
 export class ConnectionRequestHandler implements Handler {
   private connectionService: ConnectionService
   private agentConfig: AgentConfig
+  private mediationRecipientService: MediationRecipientService
   public supportedMessages = [ConnectionRequestMessage]
 
-  public constructor(connectionService: ConnectionService, agentConfig: AgentConfig) {
+  public constructor(
+    connectionService: ConnectionService,
+    agentConfig: AgentConfig,
+    mediationRecipientService: MediationRecipientService
+  ) {
     this.connectionService = connectionService
     this.agentConfig = agentConfig
+    this.mediationRecipientService = mediationRecipientService
   }
 
   public async handle(messageContext: HandlerInboundMessage<ConnectionRequestHandler>) {
     if (!messageContext.connection) {
       throw new AriesFrameworkError(`Connection for verkey ${messageContext.recipientVerkey} not found!`)
     }
 
-    await this.connectionService.processRequest(messageContext)
+    let routing: Routing | undefined
 
-    if (messageContext.connection?.autoAcceptConnection ?? this.agentConfig.autoAcceptConnections) {
-      const { message } = await this.connectionService.createResponse(messageContext.connection.id)
-      return createOutboundMessage(messageContext.connection, message)
+    // routing object is required for multi use invitation, because we're creating a
+    // new keypair that possibly needs to be registered at a mediator
+    if (messageContext.connection.multiUseInvitation) {
+      const mediationRecord = await this.mediationRecipientService.discoverMediation()
+      routing = await this.mediationRecipientService.getRouting(mediationRecord)
+    }
```

Comment:
Maybe not. But I would rather take all edge cases into account for these things. Otherwise we need good documentation on what features do and don't work together :)